Library/Homebrew/extend/os/linux/requirements/java_requirement.rb

```diff
@@ -0,0 +1,24 @@
+require "language/java"
+
+class JavaRequirement < Requirement
+  fatal true
+  default_formula "jdk"
+  download "http://www.oracle.com/technetwork/java/javase/downloads/index.html"
+
+  satisfy build_env: false do
+    jdk = Formula["jdk"]
+    @java_home = jdk.opt_prefix if jdk && jdk.installed?
+    next quiet_system "java", "-version"
```

Comment:
Note that it is a known issue with the Linux code that it only validates that Java exists, not that it meets any version requirements.

I'd like to fix that at some point, because people on older but still-supported distros like 14.04 are, IIRC, likely using OpenJDK 7 and more and more formulae are starting to require Java 8.